gp-templates/helper-functions.php

```diff
@@ -514,3 +524,23 @@ function gp_translations_bulk_actions_toolbar( $bulk_action, $can_write, $transl
 </form>
 <?php
 }
+
+/**
+ * Determine if the current element should be skipped.
+ *
+ * @since 4.0.0
+ *
+ * @param int   $key           The key of the current element.
+ * @param array $split_element The split element.
+ *
+ * @return bool
+ */
+function should_skip_element( int $key, array $split_element ): bool {
+	if ( $key > 0 &&
+		( '<' === substr( $split_element[ $key - 1 ], -1 ) ||
+			( '"' === substr( $split_element[ $key - 1 ], -1 ) && ! ( 'alt="' === substr( $split_element[ $key - 1 ], -5 ) || 'title="' === substr( $split_element[ $key - 1 ], -7 ) ) ) ||
+			'</' === substr( $split_element[ $key - 1 ], -2 ) ) ) {
+		return true;
+	}
+	return false;
```

Comment:
I am a fan of keeping nesting levels low and ifs simple. You could refactor it to this:
```suggestion
function should_skip_element( int $key, array $split_element ): bool {
	if ( $key <= 0 ) {
		// We cannot skip the first element.
		return false;
	}
	$previous_element = $split_element[ $key - 1 ];
	$previous_character = substr( $previous_element, -1 );

	// Glossary word is an HTML tag because it is preceeded with a <, e.g. <strong>.
	if ( '<' === $previous_character ) {
		 return true;
	}

	// Glossary word is an HTML end tag because it is preceeded with a </, e.g. </strong>.
	if ( '</' === substr( $previous_element, -2 ) ) {
		 return true;
	}

	// If the preceeding char is a quotation mark, we might be in an HTML attribute.
	if ( '"' === $previous_character || "'" === $previous_character ) {
		// Don't skip if we're in an alt attribute.
		if ( 'alt=' === strtolower( substr( $previous_element, -5, 4 ) ) ) {
			return false;
		}

		// Don't skip if we're in a title attribute.
		if ( 'title=' === strtolower( substr( $previous_element, -7, 6 ) ) ) {
			return false;
		}

		// Skip if we're in any other attribute.
		return true;
	}

	// Don't skip in any other cases.
	return false;
```

Maybe we can also put this function inside the `map_glossary_entries_to_translation_originals` since we don't need it anywhere else and thus would not bloat the global namespace.

Reply:
Since we only need `$previous_element = $split_element[ $key - 1 ]` we could make that the function parameter as well but that would shift some complexity to the loop. We also don't have to go the `$key - 1` route but could do something like
```php
$previous_element = '';
foreach ( $singular_split as $chunk ) {
	if ( $this->should_skip_element( $previous_element ) ) {
		$previous_element = $chunk;
		continue;
	}
	// ...
	$previous_element = $chunk;
}
```

and
```php
function should_skip_element( string $previous_element ): bool {
	if ( ! $previous_element ) {
		// We cannot skip the first element.
		return false;
	}
	$previous_character = substr( $previous_element, -1 );
	// ...
}
```